src/Sentry/Internal/Hub.cs

```diff
@@ -189,8 +234,27 @@ public SentryId CaptureEvent(SentryEvent evt, Scope? scope = null)
                     evt.Contexts.Trace.ParentSpanId = linkedSpan.ParentSpanId;
                 }
 
+                // Report an error on current session if contains an exception
+                var sessionUpdate = evt.Exception is not null
```

Comment:
The edge case here is that if this was a capture event that came as a result of a manually crafted event or even deserializing an already proceeded event, this instance (`evt.Exception`) will be null but `SentryException` wouldn't be null so ideally this could be:

```suggestion
                var sessionUpdate = evt.Exception is not null || evt.SentryException?.Any() == true
```